lib/mapObjects/CGHeroInstance.cpp

```diff
@@ -103,11 +103,9 @@ ui32 CGHeroInstance::getTileCost(const TerrainTile & dest, const TerrainTile & f
 		&& !ti->hasBonusOfType(Bonus::NO_TERRAIN_PENALTY, from.terType.id()) //no special movement bonus
 		)
 	{
-		static const CSelector selectorPATHFINDING = Selector::typeSubtype(Bonus::SECONDARY_SKILL_PREMY, SecondarySkill::PATHFINDING);
-		static const std::string keyPATHFINDING = "type_"+std::to_string((si32)Bonus::SECONDARY_SKILL_PREMY)+"s_"+std::to_string((si32)SecondarySkill::PATHFINDING);
 
 		ret = VLC->heroh->terrCosts[from.terType];
```

Comment:
Really it should be possible to calculate and cache the whole tile cost by terrain type. At least what I have seen so far we only relay on source tile terrain type and separate logic for roads. So taking terrain cost by terrain type from some map should be even faster. So I might be missing something

Reply:
Terrain type will get total rewrite from me, actually I already did that.